### ISSUE TITLE
quadratic runtime for render_multiple_stars and render_multiple_galaxies

### ISSUE THREAD

**Issue by jeff-regier** [prob-ml/bliss#87]
Currently `render_multiple_galaxies` and `render_multiple_stars` have quadratic runtime in the size for the images (assuming number of light source scales with image size). For large images (e.g., one SDSS frame), this approach is intractable. The problem is that, for an SDSS frame, the current code would generate a 2-megapixel image for each of tens of thousands of light sources, even though the vast majority of each image would be zero padding.

The code below shows how to reduce the runtime to linear. Here `z` is a draw of the latent variables for each tile. The image to reconstruct is 1024 x 2048. The tiles are 2 x 2 and the padded tiles are 8 x 8. `canvas` contains the reconstructed mean.

```python
import torch

z = torch.rand((512, 1024, 3))
decoder = torch.nn.Linear(3, 8 * 8)
z_2d = z.view((-1, 3))
x_2d = decoder(z_2d)
x_4d = x_2d.view(512, 1024, 8, 8)
x_6d = x_4d.view(128, 4, 256, 4, 8, 8).permute([1, 3, 0, 4, 2, 5])
x44 = x_6d.reshape(4, 4, 128 * 8, 256 * 8)

canvas = torch.zeros(1024 + 6, 2048 + 6)
for i in range(4):
    for j in range(4):
        canvas[(2 * i):(1024 + 2 * i), (2 * j):(2048 + 2 * j)] += x44[i, j]
```
